modules/services/window-managers/i3-sway/sway.nix

```diff
@@ -263,53 +263,49 @@ let
+      "focus_wrapping ${if focus.forceWrapping then "yes" else "no"}"
+      "focus_follows_mouse ${focus.followMouse}"
+      "focus_on_window_activation ${focus.newWindow}"
+      "mouse_warping ${if focus.mouseWarping then "output" else "none"}"
+      "workspace_layout ${workspaceLayout}"
+      "workspace_auto_back_and_forth ${
         if workspaceAutoBackAndForth then "yes" else "no"
-      }
-
-      client.focused ${colorSetStr colors.focused}
-      client.focused_inactive ${colorSetStr colors.focusedInactive}
-      client.unfocused ${colorSetStr colors.unfocused}
-      client.urgent ${colorSetStr colors.urgent}
-      client.placeholder ${colorSetStr colors.placeholder}
-      client.background ${colors.background}
-
-      ${keybindingsStr {
+      }"
+      "client.focused ${colorSetStr colors.focused}"
+      "client.focused_inactive ${colorSetStr colors.focusedInactive}"
+      "client.unfocused ${colorSetStr colors.unfocused}"
+      "client.urgent ${colorSetStr colors.urgent}"
+      "client.placeholder ${colorSetStr colors.placeholder}"
+      "client.background ${colors.background}"
+      (keybindingsStr {
         keybindings = keybindingDefaultWorkspace;
         bindsymArgs =
           lib.optionalString (cfg.config.bindkeysToCode) "--to-code";
-      }}
-      ${keybindingsStr {
+      })
+      (keybindingsStr {
         keybindings = keybindingsRest;
         bindsymArgs =
           lib.optionalString (cfg.config.bindkeysToCode) "--to-code";
-      }}
-      ${keycodebindingsStr keycodebindings}
-      ${concatStringsSep "\n" (
-        # Append all of the lists together to avoid unnecessary whitespace.
-        mapAttrsToList inputStr input # inputs
-        ++ mapAttrsToList outputStr output # outputs
-        ++ mapAttrsToList seatStr seat # seats
-        ++ mapAttrsToList (modeStr cfg.config.bindkeysToCode) modes # modes
-        ++ mapAttrsToList assignStr assigns # assigns
-        ++ map barStr bars # bars
-        ++ optional (gaps != null) gapsStr # gaps
-        ++ map floatingCriteriaStr floating.criteria # floating
-        ++ map windowCommandsStr window.commands # window commands
-        ++ map startupEntryStr startup # startup
-        ++ map workspaceOutputStr workspaceOutputAssign # custom mapping
-      )}
-    ''
+      })
+      (keycodebindingsStr keycodebindings)
+    ] ++ mapAttrsToList inputStr input
+      ++ mapAttrsToList outputStr output # outputs
+      ++ mapAttrsToList seatStr seat # seats
+      ++ mapAttrsToList (modeStr cfg.config.bindkeysToCode) modes # modes
+      ++ mapAttrsToList assignStr assigns # assigns
+      ++ map barStr bars # bars
+      ++ optional (gaps != null) gapsStr # gaps
+      ++ map floatingCriteriaStr floating.criteria # floating
+      ++ map windowCommandsStr window.commands # window commands
+      ++ map startupEntryStr startup # startup
+      ++ map workspaceOutputStr workspaceOutputAssign # custom mapping
+      ++ [ "\n" ])
```

Comment:
This is a bit hacky, but otherwise exec is on the same line as the last workspaceOutputAssign... Maybe you have a better solution?

Reply:
You could try moving the `concatStringsSep` above the `if` statement to encompass both cases.

Reply:
Thanks you for the hint, it brought me on the right direction I think - I like the new solution a lot more ;)